frontend/src/app.js

```diff
@@ -194,3 +194,34 @@ document.getElementById('menu-button').onclick = function toggleMenu() {
     element.classList.toggle('hidden');
   }
 };
+
+const mobFilterButton = document.querySelector('#mobFilterButton')
+const filters = document.querySelector('#filters')
+const backdrop  = document.querySelector('.backdrop')
+const filterClose  = document.querySelector('.filter-close')
+const applyFilter  = document.querySelector('.apply-filter-btn')
+
+mobFilterButton.onclick = () =>{
+  filters.classList.toggle('translate-x-0')
+  filters.classList.toggle('-translate-x-96')
+  backdrop.classList.toggle('hidden')
+  backdrop.classList.toggle('block')
+}
+filterClose.onclick = () =>{
+  filters.classList.toggle('translate-x-0')
+  filters.classList.toggle('-translate-x-96')
+  backdrop.classList.toggle('hidden')
+  backdrop.classList.toggle('block')
+}
+applyFilter.onclick = () =>{
+  filters.classList.toggle('translate-x-0')
+  filters.classList.toggle('-translate-x-96')
+  backdrop.classList.toggle('hidden')
+  backdrop.classList.toggle('block')
+}
+backdrop.onclick = () =>{
+  filters.classList.toggle('translate-x-0')
+  filters.classList.toggle('-translate-x-96')
+  backdrop.classList.toggle('hidden')
+  backdrop.classList.toggle('block')
```

Comment:
Toggling `hidden` is fine, as it's semantic. However I don't like toggling the other classes in JavaScript, because it spreads the styling. Can we consider any other approach to toggle only semantic classes in JS, which would lead to effectively applying additional Tailwind utilities? I think of a custom variant (https://tailwindcss.com/docs/hover-focus-and-other-states#creating-custom-variants, https://tailwindcss.com/docs/plugins#basic-variants). I.e., I'd toggle `activated` class on the filter and backbrop, and use custom variant to style these elements like this: `hidden -translate-x-96 activated:translate-x-0 activated:block`. Do you like the idea?

Reply:
By the way, I've just noticed that the click handlers are identical, all 4! Let's don't repeat ourselves.